vlog4j-rdf/src/main/java/org/semanticweb/vlog4j/rdf/RdfValueToTermConverter.java

```diff
@@ -41,25 +44,32 @@ static Term rdfValueToTerm(final Value value) {
 		} else if (value instanceof Literal) {
 			return rdfLiteralToConstant((Literal) value);
 		} else if (value instanceof URI) {
-			return rdfURItoConstant((URI) value);
+			return rdfUriToConstant((URI) value);
 		} else {
 			throw new RuntimeException("Unknown value type: " + value.getClass());
 		}
 	}
 
 	static Term rdfBlankNodeToBlank(final BNode bNode) {
 		// IDs are generated to be unique in every Model.
-		return new BlankImpl(bNode.getID());
+		return new NamedNullImpl(bNode.getID());
 	}
 
-	static Term rdfURItoConstant(final URI uri) {
+	static Term rdfUriToConstant(final URI uri) {
 		final String escapedURIString = NTriplesUtil.escapeString(uri.toString());
-		return new ConstantImpl(escapedURIString);
+		return new AbstractConstantImpl(escapedURIString);
 	}
 
 	static Term rdfLiteralToConstant(final Literal literal) {
-		final String normalizedStringValueLiteral = buildNormalizedStringValue(literal);
-		return new ConstantImpl(normalizedStringValueLiteral);
+		final URI datatype = literal.getDatatype();
```

Comment:
method `static String buildNormalizedStringValue(final Literal literal)` is not used anywhere now. It should be removed.